File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/anmodning/AnmodningsperiodeGetDto.java
package no.nav.melosys.tjenester.gui.dto.anmodning;

import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

import no.nav.melosys.domain.Anmodningsperiode;

public final class AnmodningsperiodeGetDto {
    private List<AnmodningsperiodeLesDto> anmodningsperioder;

    public AnmodningsperiodeGetDto() {
    }

    private AnmodningsperiodeGetDto(Collection<Anmodningsperiode> anmodningsperioder) {
            this.anmodningsperioder = anmodningsperioder.stream()
            .map(AnmodningsperiodeLesDto::av)
            .collect(Collectors.toList());
    }

    public static AnmodningsperiodeGetDto av(Collection<Anmodningsperiode> anmodningsperioder) {
        return new AnmodningsperiodeGetDto(anmodningsperioder);
    }

    public void setAnmodningsperioder(List<AnmodningsperiodeLesDto> anmodningsperioder) {
        this.anmodningsperioder = anmodningsperioder;
    }

    public List<AnmodningsperiodeLesDto> getAnmodningsperioder() {
        return anmodningsperioder;
    }
}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/dto/anmodning/AnmodningsperiodeGetDto.kt
package no.nav.melosys.tjenester.gui.dto.anmodning

import no.nav.melosys.domain.Anmodningsperiode
import java.util.stream.Collectors

class AnmodningsperiodeGetDto {
    var anmodningsperioder: List<AnmodningsperiodeLesDto>? = null

    private constructor(anmodningsperioder: Collection<Anmodningsperiode>) {
        this.anmodningsperioder = anmodningsperioder.stream()
            .map { anmodningsperiode: Anmodningsperiode? ->
                AnmodningsperiodeLesDto.av(
                    anmodningsperiode!!
                )
            }
            .collect(Collectors.toList())
    }

    companion object {
        @JvmStatic
        fun av(anmodningsperioder: Collection<Anmodningsperiode>): AnmodningsperiodeGetDto {
            return AnmodningsperiodeGetDto(anmodningsperioder)
        }
    }
}
